src/commonMain/kotlin/org/mifek/wfc/datatypes/Directions2D.kt

```diff
@@ -0,0 +1,24 @@
+package org.mifek.wfc.datatypes
+
+enum class Directions2D(private val value: Int) {
+    NORTH(0),
+    EAST(1),
+    SOUTH(2),
+    WEST(3);
+
+    fun toInt(): Int {
+        return value
+    }
+
+    companion object {

```

Comment:
I think this is neater
```suggestion
companion object {
        private val allValues = values();
        fun fromInt(value: Int): Directions2D {
            return allValues[value];
        }
    }
```